frontend/packages/yki/src/tests/jest/utils/examSession.test.ts

```diff
@@ -0,0 +1,247 @@
+        })
+      ).toEqual(-1);
+
+      expect(
+        ExamSessionUtils.compareExamSessions(
+          {
+            ...baseExamSession,
+            queue_full: true,
+          },
+          baseExamSession
+        )
+      ).toEqual(1);
+    });
+
+    it('should prioritise exam sessions with earlier session date', () => {
+      expect(
+        ExamSessionUtils.compareExamSessions(
+          {
+            ...baseExamSession,
+            session_date: dayjs('2098-12-31'),
+          },
+          baseExamSession
+        )
+      ).toEqual(-1);
+
+      expect(
+        ExamSessionUtils.compareExamSessions(baseExamSession, {
+          ...baseExamSession,
+          session_date: dayjs('2098-12-31'),
+        })
+      ).toEqual(1);
+    });
+
+    it('should prioritise exam sessions without ended registration period', () => {
+      expect(
+        ExamSessionUtils.compareExamSessions(baseExamSession, {
+          ...baseExamSession,
+          registration_end_date: dayjs('2021-01-01'),
+        })
+      ).toEqual(-1);
+
+      expect(
+        ExamSessionUtils.compareExamSessions(
+          {
+            ...baseExamSession,
+            registration_end_date: dayjs('2021-01-01'),
+          },
+          baseExamSession
+        )
+      ).toEqual(1);
+    });
+
+    it('should prioritise comparators', () => {
+      // language code > room
+      expect(
+        ExamSessionUtils.compareExamSessions(
+          {
+            ...baseExamSession,
+            language_code: ExamLanguage.DEU,
+            participants: baseExamSession.max_participants,
+          },
+          baseExamSession
+        )
+      ).toEqual(-1);
+
+      // room > queue fullness
+      expect(
+        ExamSessionUtils.compareExamSessions(
+          {
+            ...baseExamSession,
+            participants: baseExamSession.max_participants,
+          },
+          {
+            ...baseExamSession,
+            queue_full: true,
+          }
+        )
+      ).toEqual(1);
+
+      // queue fullness > earlier session date
+      expect(
+        ExamSessionUtils.compareExamSessions(
+          {
+            ...baseExamSession,
+            queue_full: true,
+            session_date: dayjs('2098-12-31'),
+          },
+          baseExamSession
+        )
+      ).toEqual(1);
+    });
+
+    // earlier session date > ended registration
+    // participants set as max_participants for both to avoid es1 to be considered full and es2 not
+    expect(
+      ExamSessionUtils.compareExamSessions(
+        {
+          ...baseExamSession,
+          participants: baseExamSession.max_participants,
+          session_date: dayjs('2098-12-31'),
+          registration_end_date: dayjs('2021-01-01'),
+        },
+        {
+          ...baseExamSession,
+          participants: baseExamSession.max_participants,
+        }
+      )
+    ).toEqual(-1);
+  });
+
+  it('should treat exam sessions with post admission similarly to ones without', () => {
```

Comment:
Meni hetki aikaa tajuta mitä tällä testikeissillä haetaan - ymmärtääkseni siis sitä että varsinaiseen ja jälki-ilmoittautumiseen liittyviä ilmoittautumisaikoja sekä osallistujamääriä käsitellään samanarvoisesti vertailulogiikassa?

Jos tämä oli oikea tulkinta, niin testin kuvaus voisi ehkä viitata tähän selkeämmin.
Vaikkapa jotain luokkaa `it('should value availability of regular and post admission equally')`?

Reply:
Joo, on parempi kuvaus testille tämä ehdottamasi